### PR TITLE
[docs] Update IIS README to use documentation guidelines

### PR DESCRIPTION
**Pull request by colleenmcginnis**
## Proposed commit message

Update README to use documentation guidelines

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] ~~I have verified that all data streams collect metrics or logs.~~
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] ~~I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~~

## Author's Checklist

- [x] @colleenmcginnis Initial draft to align with the [documentation guidelines](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md)
- [x] @SubhrataK to provide an initial review of the structure
- [x] @colleenmcginnis address initial feedback, remove draft status
- [x] @elastic/obs-infraobs-integrations technical review 
- [x] @colleenmcginnis address additional feedback, merge

## How to test this PR locally

View [the README](https://github.com/colleenmcginnis/integrations/blob/iis-setup/packages/iis/docs/README.md).

## Related issues

Related to https://github.com/elastic/observability-docs/issues/3270

### CODE REVIEW

File: packages/iis/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.15.2"
+  changes:
+    - description: Update README to use documentation guidelines
+      type: enhancement
+      link: TBD
```
Comment (colleenmcginnis) [inline packages/iis/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/8241
```

**Review by SubhrataK — APPROVED** [review]
Looks good!

File: packages/iis/_dev/build/docs/README.md [hunk 2]
```diff
@@ -78,13 +76,45 @@ The fields reported are:
...
-{{fields "error"}}
\ No newline at end of file
+{{fields "error"}}
+
+
+## Metrics
+
+### webserver
+
+The `webserver` dataset allows users to retrieve aggregated metrics for the entire webserver.
```
[the remark this answers is not in the document]
Comment (shmsr) [reply packages/iis/_dev/build/docs/README.md:94]: nit

**Review by shmsr — APPROVED** [review]
Thank you for improving the documentation 🎉

File: packages/iis/_dev/build/docs/README.md [hunk 3]
```diff
@@ -1,45 +1,42 @@
...
-
-The fields reported are:
+## Requirements
 
-{{fields "website"}}
+You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it.
+You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your own hardware.
 
-### application_pool
-This dataset will collect metrics of specific application pools, users can configure which websites they want to monitor, else, all are considered.
```
Comment (ishleenk17) [inline packages/iis/_dev/build/docs/README.md:36]: Dataset and datastream usage is done interchangeably. 
Is that the practice we are following for in documentation guideline ?

Comment (colleenmcginnis) [reply packages/iis/_dev/build/docs/README.md:36]: In the [integration documentation guidelines](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md) we use "data stream". I updated this PR to consistently use "data stream".

**Pull request merged**